lib/route/src/Obelisk/Route.hs

```diff
@@ -398,25 +399,24 @@ instance Monad parse => Category (EncoderImpl parse) where
     , _encoderImpl_encode = _encoderImpl_encode f . _encoderImpl_encode g
     }
 
-instance Monad parse => PFunctor (,) (EncoderImpl parse) (EncoderImpl parse) where
+instance (Monad parse, Bitraversable p, Bifunctor p (->) (->) (->)) => PFunctor p (EncoderImpl parse) (EncoderImpl parse) where
   first f = bimap f id
-instance Monad parse => QFunctor (,) (EncoderImpl parse) (EncoderImpl parse) where
+instance (Monad parse, Bitraversable p, Bifunctor p (->) (->) (->)) => QFunctor p (EncoderImpl parse) (EncoderImpl parse) where
   second g = bimap id g
-instance Monad parse => Bifunctor (,) (EncoderImpl parse) (EncoderImpl parse) (EncoderImpl parse) where
+instance (Monad parse, Bitraversable p, Bifunctor p (->) (->) (->)) => Bifunctor p (EncoderImpl parse) (EncoderImpl parse) (EncoderImpl parse) where
```

Comment:
I do wonder whether there can be a `Bifunctor p (EncoderImpl parse) (EncoderImpl parse) (EncoderImpl parse)` that does anything different than `bimap` over `p a b` in the forward direction since it must be able to encode `forall a b`
As for it can do anything besides `bitraverse` in the backward direction, that equally applies to the `Functor (EncoderImpl parse) (EncoderImpl parse)` instance